### PR TITLE
Add support for Amazon Corretto

### PR DESCRIPTION
**Pull request by unicolet**
adds support for Amazon Corretto. For now we fetch the artifact from the upstream source, but it might be more reliable to self host it on our S3 bucket.

### CODE REVIEW

**Comment by SISheogorath** [comment]
How is the update of the hosted artifact and key done? Manually? Automated?

Especially for the key we might want to update it on a regular basis to make sure it's not revoked.

**Comment by unicolet** [comment]
@SISheogorath I'm mostly worried about Amazon pulling older releases from their CDN (or deploying to another CDN entirely). And since `8.202.08.2` is their first GA that's hard to tell

How can we check if they key was revoked?

**Comment by SISheogorath** [comment]
When revoking a key, the key owner uploads a signed revocation of the key onto keyservers which then "spread the news". GPG contains a service called dirmngr which revisits keys on a regular basis and updates them from the remote to receive the latest signatures (for WoT use) and revocations.

**Comment by unicolet** [comment]
@SISheogorath I have changed the key retrieval logic to fetch from a server. Let me know if it's good enough now, thanks :-)

File: bin/install [hunk 1]
```diff
@@ -82,6 +89,23 @@ get_java() {
...
+    
+	if [ "$signature" != "" ] && [ "$sigkey" != "" ]; then
+		curl -fLC - \
+			--progress-bar --silent \
+			--retry 3 --retry-delay 3 \
+			-b oraclelicense=a \
+			-o $destdir/${variant}.sig $signature
+		
+		# gpg servers are known not to be reliable
+		gpg --recv $sigkey || gpg --recv $sigkey || gpg --recv $sigkey
```
Comment (SISheogorath) [inline bin/install:102]: :joy: LOL :+1:

File: bin/install [hunk 2]
```diff
@@ -82,6 +89,23 @@ get_java() {
 		--retry 3 --retry-delay 3 \
 		-b oraclelicense=a \
 		-o $destdir/$variant $url
+    
+	if [ "$signature" != "" ] && [ "$sigkey" != "" ]; then
```
Comment (SISheogorath) [inline bin/install:94]: Do we want to make signature verification optional?

File: bin/install [hunk 3]
```diff
@@ -57,12 +57,19 @@ get_java() {
...
+	local sigkey=
 	local base=http://download.oracle.com/otn-pub/java/jdk/
 	local oracle_version=$(echo $version | sed -e 's/\./u/g')
 
 	local s3base=https://s3-eu-west-1.amazonaws.com/repo.proemion.com
 
 	case $version in
+		corretto-8.202.08.2)
+			url=https://d2znqt9b1bc64u.cloudfront.net/amazon-corretto-8.202.08.2-linux-x64.tar.gz 
+			signature=https://d2znqt9b1bc64u.cloudfront.net/amazon-corretto-8.202.08.2-linux-x64.tar.gz.sig
```
Comment (SISheogorath) [inline bin/install]: Just as recommendation, we may want to add a version variable here? Just to make updates easier to keep signatures and downloaded version in sync

File: bin/install (hunk 2 above)
Comment (unicolet) [reply bin/install:94]: @SISheogorath only for the other (older) distributions that do not set the vars

File: bin/install (hunk 3 above)
Comment (unicolet) [reply bin/install]: @SISheogorath done :-)

File: bin/install (hunk 2 above)
Comment (SISheogorath) [reply bin/install:94]: I see.

**Review by SISheogorath — APPROVED** [review]

**Pull request merged**